test/colony-network-recovery.js

```diff
@@ -0,0 +1,498 @@
+      assert.equal(nNodes.toNumber(), 7);
+    });
+
+    process.env.SOLIDITY_COVERAGE // eslint-disable-line no-unused-expressions
+      ? it.skip
+      : it("miner should be able to correctly interpret historical reputation logs replaced during recovery mode", async () => {
+          await giveUserCLNYTokensAndStake(colonyNetwork, accounts[5], toBN(10).pow(toBN(18)));
+          await miningClient.saveCurrentState();
+          const startingHash = await miningClient.getRootHash();
+
+          const newMiningClient = new ReputationMiner({
+            loader: contractLoader,
+            minerAddress: accounts[5],
+            realProviderPort: REAL_PROVIDER_PORT,
+            useJsTree: true
+          });
+          await newMiningClient.initialise(colonyNetwork.address);
+
+          const tokenArgs = getTokenArgs();
+          const token = await Token.new(...tokenArgs);
+          const { logs } = await colonyNetwork.createColony(token.address);
+          const { colonyAddress } = logs[0].args;
+
+          await token.setOwner(colonyAddress);
+          const colony = await IColony.at(colonyAddress);
+
+          await colony.mintTokens(1000000000000000);
+          await colony.bootstrapColony([accounts[0]], [1000000000000000]);
+
+          let addr = await colonyNetwork.getReputationMiningCycle.call(false);
+          let repCycle = await IReputationMiningCycle.at(addr);
+
+          await miningClient.addLogContentsToReputationTree();
+          await forwardTime(3600, this);
+          await miningClient.submitRootHash();
+
+          addr = await colonyNetwork.getReputationMiningCycle.call(true);
+          repCycle = await IReputationMiningCycle.at(addr);
+          await repCycle.confirmNewHash(0);
+
+          await miningClient.addLogContentsToReputationTree();
+          await forwardTime(3600, this);
+          await miningClient.submitRootHash();
+
+          addr = await colonyNetwork.getReputationMiningCycle.call(true);
+          repCycle = await IReputationMiningCycle.at(addr);
+
+          const invalidEntry = await repCycle.getReputationUpdateLogEntry(5);
+          invalidEntry.amount = 0;
+
+          await repCycle.confirmNewHash(0);
+
+          const domain = await colony.getDomain(1);
+          const rootSkill = domain[0];
+          const reputationKey = makeReputationKey(colony.address, rootSkill.toNumber(), accounts[0]);
+          const originalValue = miningClient.reputations[reputationKey].slice(2, 66);
+          assert.equal(parseInt(originalValue, 16), 1000000000000000);
+
+          await colonyNetwork.enterRecoveryMode();
+
+          await colonyNetwork.setReplacementReputationUpdateLogEntry(
+            addr,
+            5,
+            invalidEntry.user,
+            invalidEntry.amount,
+            invalidEntry.skillId,
+            invalidEntry.colony,
+            invalidEntry.nUpdates,
+            invalidEntry.nPreviousUpdates
+          );
```

Comment:
This sets a what is now a destructed ReputationMiningCycle entry, for the purposes of correct future client `sync` calls